factory/grpcServerFactory.go

```diff
@@ -0,0 +1,49 @@
+package factory
+
+import (
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+
+	"github.com/multiversx/mx-chain-ws-connector-template-go/config"
+	data "github.com/multiversx/mx-chain-ws-connector-template-go/data/hyperOutportBlocks"
+	"github.com/multiversx/mx-chain-ws-connector-template-go/process"
+	"github.com/multiversx/mx-chain-ws-connector-template-go/service/hyperOutportBlock"
+)
+
+type GRPCServer struct {
+	server *grpc.Server
+	config config.GRPCConfig
+}
+
+// NewServer instantiates the underlying grpc server handling rpc requests.
+func NewServer(config config.GRPCConfig, blocksHandler process.GRPCBlocksHandler) *GRPCServer {
+	s := grpc.NewServer()
+
+	service := hyperOutportBlock.NewService(blocksHandler)
+	data.RegisterHyperOutportBlockServiceServer(s, service)
+	reflection.Register(s)
+
+	return &GRPCServer{s, config}
+}
+
+// Start will start the grpc server on the configured URL.
+func (s *GRPCServer) Start() error {
+	lis, err := net.Listen("tcp", s.config.URL)
```

Comment:
move this to a separate component, and add the goroutine into this component, we usually use a pattern like this one here https://github.com/multiversx/mx-chain-notifier-go/blob/main/process/publisher.go#L48